### DISCUSSION TITLE
sliding window

### DISCUSSION THREAD

**Discussion by lxzan** [klauspost/compress#849] [General]
I see that the master branch has a new NewWriterWindow function, can I use it to implement the full WebSocket RFC7692 compression extension?

Thanks for your great work

**Comment by klauspost** [comment]
@lxzan It does indeed seem so.

**Comment by lxzan** [comment]
I'm going to have to add a bunch of ugly if else to support the client_max_window_bits/server_max_window_bits parameters of the websocket permessage-deflate extension :(
